### PR TITLE
operator/identitygc: Disable identitygc when Operator manages CID

### PR DESCRIPTION
**Pull request by ovidiutirla**
`EnableOperatorManageCIDs` enables operator to manage CID by running a CID controller. If enabled, Identity GC cell is then disabled because CID controller takes care of garbage collection.

Related https://github.com/cilium/cilium/issues/27752

Draft full implementation: https://github.com/cilium/cilium/pull/33204

### CODE REVIEW

File: operator/identitygc/cell.go
Comment (pippolo84) [inline operator/identitygc/cell.go:87]: The management of the new option in this cell is correct, but we need to add an operator option to allow the user to set this one. Look at how `IdentityAllocationMode` is initialized in operator/cmd/root.go from the related option in daemonConfig for an example.

**Review by pippolo84 — CHANGES_REQUESTED** [review]

File: operator/identitygc/cell.go
Comment (ovidiutirla) [reply operator/identitygc/cell.go:87]: Hey @pippolo84 thank you for your review!

We'll add this in a later PR, by default this is marked as false. I intentionally not added that here so we don't have to worry about mistakenly enabling this until the full implementation is merged.

**Review by pippolo84 — APPROVED** [review]

**Comment by joestringer** [comment]
Given the detailed discussion going on about the implementation details of how CiliumIdentity garbage collection will happen in future going on in https://github.com/cilium/cilium/pull/33380#discussion_r1663971780, I think this PR was prematurely merged cc @pippolo84 @ovidiutirla @pchaigno . The assumption here that the GC implementations will diverge is clearly a bit contentious.

I'll hold off on reverting for this for the moment because I think it's useful for us to continue the discussion and agree on the path forward. I see value in the ideas being proposed, the question will be quite how we get there.

**Pull request merged**
